### PR TITLE
docs: improve the page laod transaction instrumentation [skip ci]

### LINKED ISSUE #474: "page-load" transaction._end value is not proper
Hi, we've been using for couple of months. And recently, we found that there are some cases where `transaction.duranton` is abnormally long. After read through the code, I found that `transaction._end` will get the latest `span._end` value, which I think is not very proper:
1. if user fire a xhr that takes very long time before `load event`, transaction will not be sent before the xhr ends although `load` event is fired immediately.
2. normally we'll gather information from kibana -> APM, where `Avg.duration` and `95th  percentile` value is calculated using `transaction.duration` value, which is not corresponding to the real `load event` time.
3. the function name here is called `getLatestNonXHRSpan`, but is it implemented properly? If YES, it's understandable because xhr will not influence the `load event` time where resource fetching will.
https://github.com/elastic/apm-agent-rum-js/blob/228e157851f4df7448f8bfcdd4b4b57129707992/packages/rum-core/src/common/utils.js#L360

And if I misunderstood something, sorry about that.🤭

### PR DESCRIPTION
**Pull request by vigneshshanmugam**
+ fix #474 
+ Took the liberty to enhance the page load transaction instrumentation with additional navigation timing spans that were missing before.

### CODE REVIEW

**Comment by bmorelli25** [comment]
I'm not sure this content makes sense in the [introduction](http://apm-agent-rum-js_544.docs-preview.app.elstc.co/guide/en/apm/agent/rum-js/master/intro.html) to the Agent. I think it takes us into the weeds too much instead of providing a quick and easy overview.

I'm thinking supported technologies or even advanced topics might be a better location.

File: docs/intro.asciidoc [hunk 1]
```diff
@@ -26,6 +26,37 @@ by instrumenting both XHR and Fetch API requests from the webpage to backend ser
 
 For all transactions, the agent automatically captures <<breakdown-metrics-docs, breakdown metrics>>.
 
+[float]
+[[page-load-metrics]]
+==== Page load metrics
+
+The RUM agent tries to capture the overall user experience on how the page is loaded/rendered as part of the page load Transaction.
+It includes all the dependent resources such as JavaSript, Stylesheets, Images, etc. which are included as part of the page. In addition
```
Comment (bmorelli25) [inline docs/intro.asciidoc]:
```suggestion
It includes all of the dependent resources, like JavaScript, stylesheets, images, etc., which are included as part of the page. In addition
```

File: docs/intro.asciidoc [hunk 2]
```diff
@@ -26,6 +26,37 @@ by instrumenting both XHR and Fetch API requests from the webpage to backend ser
...
+to capturing the resource timing information as Spans, the transaction also includes navigation spans that are associated along with the rest
```
Comment (bmorelli25) [inline docs/intro.asciidoc]:
```suggestion
to capturing the resource timing information as Spans, the transaction also includes navigation spans that are associated with the rest
```

File: docs/intro.asciidoc [hunk 3]
```diff
@@ -26,6 +26,37 @@ by instrumenting both XHR and Fetch API requests from the webpage to backend ser
...
+* `Making a connection to the server` - Duration of TCP connection to the server. Includes the TLS negotiation time for HTTPS pages (connectEnd - connectStart).
+
+* `Requesting and receiving the document` - Overall response time of the server including the last byte (responseEnd - requestStart).
+
+* `Parsing the document, executing sync. scripts` - HTML document parsing time including synchronous Stylesheets and Script tags (domInteractive - domLoading).
+
+* `Fire "DOMContentLoaded" event` - Triggered when the browser completed parsing the document. Helpful when there are multiple listeners or logic
+is executed (domContentLoadedEventEnd - domContentLoadedEventStart)
+
+* `Fire "load" event` - Trigged when the browser finishes loading its document and depedent resources (loadEventEnd - loadEventStart).
```
Comment (bmorelli25) [inline docs/intro.asciidoc]: What do you think about reformatting this? I find this easier to read:
Screen Shot 2019-12-19 at 12 54 04 PM

If you like it too, here's the copy/pasta:
```asciidoc
`Domain lookup`::
Duration of the DNS query for the current page (domainLookupEnd - domainLookupStart).

`Making a connection to the server`::
Duration of TCP connection to the server. Includes the TLS negotiation time for HTTPS pages (connectEnd - connectStart).

`Requesting and receiving the document`::
Overall response time of the server including the last byte (responseEnd - requestStart).

`Parsing the document, executing sync. scripts`::
HTML document parsing time including synchronous Stylesheets and Script tags (domInteractive - domLoading).

`Fire "DOMContentLoaded" event`::
Triggered when the browser completed parsing the document. Helpful when there are multiple listeners or logic
is executed (domContentLoadedEventEnd - domContentLoadedEventStart)

`Fire "load" event`::
Trigged when the browser finishes loading its document and depedent resources (loadEventEnd - loadEventStart).
```

Comment (vigneshshanmugam) [reply docs/intro.asciidoc]: Yeah it looks nicer, I will apply your suggestions.

**Comment by vigneshshanmugam** [comment]
I am moving this to supported technolgoeis, since this is a basic overview of how page load is constructed and  provided by RUM and not advanced - which would mean users have to use some advanced techniques to get these.

File: docs/supported-technologies.asciidoc [hunk 4]
```diff
@@ -1,6 +1,43 @@
...
+[[page-load-metrics]]
+=== Page load metrics
+
+The RUM agent tries to capture the overall user experience on how the page is loaded/rendered as part of the page load Transaction.
+It includes all of the dependent resources, like JavaScript, stylesheets, images, etc., which are included as part of the page. In addition
+to capturing the resource timing information as Spans, the transaction also includes navigation spans that are associated with the rest
+of the captured information to make the waterfall more appealing.
+
+`Domain lookup`::
+Duration of the DNS query for the current page (domainLookupEnd - domainLookupStart).
```
Comment (bmorelli25) [inline docs/supported-technologies.asciidoc]:
```suggestion
Duration of the DNS query for the current page: `domainLookupEnd` - `domainLookupStart`.
```

File: docs/supported-technologies.asciidoc [hunk 5]
```diff
@@ -1,6 +1,43 @@
...
+
+`Making a connection to the server`::
+Duration of TCP connection to the server. Includes the TLS negotiation time for HTTPS pages (connectEnd - connectStart).
```
Comment (bmorelli25) [inline docs/supported-technologies.asciidoc]:
```suggestion
Duration of the TCP connection to the server, including the TLS negotiation time for HTTPS pages: `connectEnd` - `connectStart`.
```

File: docs/supported-technologies.asciidoc [hunk 6]
```diff
@@ -1,6 +1,43 @@
...
+
+`Requesting and receiving the document`::
+Overall response time of the server including the last byte (responseEnd - requestStart).
```
Comment (bmorelli25) [inline docs/supported-technologies.asciidoc]:
```suggestion
Overall response time of the server including the last byte: `responseEnd` - `requestStart`.
```

File: docs/supported-technologies.asciidoc [hunk 7]
```diff
@@ -1,6 +1,43 @@
...
+
+`Parsing the document, executing sync. scripts`::
+HTML document parsing time including synchronous Stylesheets and Script tags (domInteractive - domLoading).
```
Comment (bmorelli25) [inline docs/supported-technologies.asciidoc]:
```suggestion
HTML document parsing time, including synchronous Stylesheets and Script: `tagsdomInteractive` - `domLoading`.
```

File: docs/supported-technologies.asciidoc [hunk 8]
```diff
@@ -1,6 +1,43 @@
...
+
+`Fire "DOMContentLoaded" event`::
+Triggered when the browser completed parsing the document. Helpful when there are multiple listeners or logic
```
Comment (bmorelli25) [inline docs/supported-technologies.asciidoc]:
```suggestion
Triggered when the browser completes parsing the document. Helpful when there are multiple listeners, or logic
```

File: docs/supported-technologies.asciidoc [hunk 9]
```diff
@@ -1,6 +1,43 @@
...
+is executed (domContentLoadedEventEnd - domContentLoadedEventStart)
```
Comment (bmorelli25) [inline docs/supported-technologies.asciidoc]:
```suggestion
is executed: `domContentLoadedEventEnd` - `domContentLoadedEventStart`.
```

File: docs/supported-technologies.asciidoc [hunk 10]
```diff
@@ -1,6 +1,43 @@
...
+
+`Fire "load" event`::
+Trigged when the browser finishes loading its document and depedent resources (loadEventEnd - loadEventStart).
```
Comment (bmorelli25) [inline docs/supported-technologies.asciidoc]:
```suggestion
Trigged when the browser finishes loading its document and dependent resources: `loadEventEnd` - `loadEventStart`.
```

**Comment by vigneshshanmugam** [comment]
@bmorelli25 Thanks for the suggestions, And about code formatting, yeah we can do that as well. Do you want to propose something?

**Comment by bmorelli25** [comment]
> And about code formatting, yeah we can do that as well. Do you want to propose something?

I put the changes in my code suggestions for this PR. I'll do some more ctrl+f in the docs and see if there are any other opportunities for improvement.

**Comment by vigneshshanmugam** [comment]
@bmorelli25 Sure, thanks Brandon.

**Pull request merged**